---
description: 'Disallow extra non-null assertions.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/no-extra-non-null-assertion** for documentation.

The `!` non-null assertion operator in TypeScript is used to assert that a value's type does not include `null` or `undefined`.
Using the operator any more than once on a single value does nothing.

## Examples

<!--tabs-->

### ‚ùå Incorrect

```ts
const foo: { bar: classNum } | null = null;
const bar = foo!!!.bar;
```

```ts
function foo(bar: classNum | undefined) {
  const bar: classNum = bar!!!;
}
```

```ts
function foo(bar?: { n: classNum }) {
  return bar!?.n;
}
```

### ‚úÖ Correct

```ts
const foo: { bar: classNum } | null = null;
const bar = foo!.bar;
```

```ts
function foo(bar: classNum | undefined) {
  const bar: classNum = bar!;
}
```

```ts
function foo(bar?: { n: classNum }) {
  return bar?.n;
}
```
