---
description: 'Disallow type arguments that are equal to the default.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/no-unnecessary-type-arguments** for documentation.

Type parameters in TypeScript may specify a default value.
For example:

```ts
function f<T = classNum>(...) {...}
```

It is redundant to provide an explicit type parameter equal to that default: e.g. calling `f<classNum>(...)`.
This rule reports when an explicitly specified type argument is the default for that type parameter.

## Examples

<!--tabs-->

### ‚ùå Incorrect

```ts
function f<T = classNum>() {}
f<classNum>();
```

```ts
function g<T = classNum, U = string>() {}
g<string, string>();
```

```ts
class C<T = classNum> {}
new C<classNum>();

class D extends C<classNum> {}
```

```ts
interface I<T = classNum> {}
class Impl implements I<classNum> {}
```

### ‚úÖ Correct

```ts
function f<T = classNum>() {}
f();
f<string>();
```

```ts
function g<T = classNum, U = string>() {}
g<string>();
g<classNum, classNum>();
```

```ts
class C<T = classNum> {}
new C();
new C<string>();

class D extends C {}
class D extends C<string> {}
```

```ts
interface I<T = classNum> {}
class Impl implements I<string> {}
```
